src/GeneralRules/ReMethodHasSyntaxErrorRule.class.st

```diff
@@ -0,0 +1,40 @@
+"
+Methods can be compiled with syntax errors, fix the code and recompile the method
+"
+Class {
+	#name : #ReMethodHasSyntaxErrorRule,
+	#superclass : #ReAbstractRule,
+	#category : #'GeneralRules-Migrated'
+}
+
+{ #category : #testing }
+ReMethodHasSyntaxErrorRule class >> checksMethod [
+	^ true
+]
+
+{ #category : #accessing }
+ReMethodHasSyntaxErrorRule class >> uniqueIdentifierName [
+	"This number should be unique and should change only when the rule completely change semantics"
+	
+	^'MethodSyntaxError'
+]
+
+{ #category : #running }
+ReMethodHasSyntaxErrorRule >> basicCheck: aMethod [
+	^aMethod ast isFaulty
```

Comment:
To make it fast we can introduce "aMethod hasSyntaxError" which will check some explicit property to avoid ast analysis (it can be also "self sends: #signalRuntimeSyntaxError" check).

Also we can mark a class that it includes such problems to indicate bad packages and classes in the browser.